website/content/en/docs/building-operators/golang/webhook.md

```diff
@@ -0,0 +1,288 @@
+│   │   ├── controller_manager_config.yaml
+│   │   ├── kustomization.yaml
+│   │   └── manager.yaml
+│   ├── manifests
+│   │   └── kustomization.yaml
+│   ├── prometheus
+│   │   ├── kustomization.yaml
+│   │   └── monitor.yaml
+│   ├── rbac
+│   │   ├── auth_proxy_client_clusterrole.yaml
+│   │   ├── auth_proxy_role.yaml
+│   │   ├── auth_proxy_role_binding.yaml
+│   │   ├── auth_proxy_service.yaml
+│   │   ├── kustomization.yaml
+│   │   ├── leader_election_role.yaml
+│   │   ├── leader_election_role_binding.yaml
+│   │   ├── memcached_editor_role.yaml
+│   │   ├── memcached_viewer_role.yaml
+│   │   ├── role_binding.yaml
+│   │   └── service_account.yaml
+│   ├── samples
+│   │   ├── cache_v1alpha1_memcached.yaml
+│   │   └── kustomization.yaml
+│   ├── scorecard
+│   │   ├── bases
+│   │   │   └── config.yaml
+│   │   ├── kustomization.yaml
+│   │   └── patches
+│   │       ├── basic.config.yaml
+│   │       └── olm.config.yaml
+│   └── webhook
+│       ├── kustomization.yaml
+│       ├── kustomizeconfig.yaml
+│       └── service.yaml
+├── controllers
+│   ├── memcached_controller.go
+│   └── suite_test.go
+├── go.mod
+├── go.sum
+├── hack
+│   └── boilerplate.go.txt
+└── main.go
+
+19 directories, 53 files
+```
+
+The scaffolded file `api/v1alpha1/memcached_webhook.go` has method signatures which need to be implemented for the validation webhook.
+
+The example memcached operator explained in the [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/) is too simple to require defaulting or additional validations,
+but as an example, we can reinforce that the default value for `spec.size` should be `3`, by adding the following logic to 
+the `Default` function (note that this is already handled by the CRD defaulting and is technically completely superfluous):
+
+```sh
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Memcached) Default() {
+	log.Info("default", "name", r.Name)
+
+	if r.Spec.Size == 0 {
+		r.Spec.Size = 3
+	}
+}
+```
+
+For validation, we can enforce that the size of the cluster follows a rule that is difficult or impossible 
+to describe with OpenAPI, for example, that the size of the cluster always remain an odd number.
+
+In order to perform this validation, we can simply add the `validateOdd` function shown below to  perform the check.
+
+```sh
+func validateOdd(n int32) error {
+	if n%2 == 0 {
+		return errors.New("Cluster size must be an odd number")
+	}
+	return nil
+}
+```
+
+At the end, let's call this method from `ValidateCreate` and `ValidateUpdate` function as shown below.
+
+```sh
+func (r *Memcached) ValidateCreate() error {
+	log.Info("validate create", "name", r.Name)
+	return validateOdd(r.Spec.Size)
+}
+
+func (r *Memcached) ValidateUpdate(old runtime.Object) error {
+	log.Info("validate update", "name", r.Name)
+	return validateOdd(r.Spec.Size)
+}
+```
+
+This function gets called whenever an object deletion happens.
```

Comment:
This line is weird, because it mentions "This" but it's unclear what it is referring to. Is it the methods above it or below it. Maybe something more like this:
```suggestion
The function below gets called whenever an object deletion happens.
```

Reply:
But if we are telling users why the function below gets called, we should also mention something similar about the `ValidateCreate` and `ValidateUpdate`.

Reply:
Modified.